lib/FusionInventory/Agent/Task/Inventory/Win32/License.pm

```diff
@@ -100,19 +90,41 @@ sub _getWmiSoftwareLicensingProducts {
             FULLNAME  => $object->{'Description'},
             NAME      => $object->{'Name'}
         };
-
-        if ($seenKey) {
-            # Update FULLNAME if seen ProductName in registry
-            $license->{FULLNAME} = encodeFromRegistry($seenKey->{'/ProductName'})
-                if $seenKey->{'/ProductName'};
-            $license->{TRIAL} = 1
-                if ($seenKey->{'/ProductNameBrand'} && $seenKey->{'/ProductNameBrand'} =~ /trial/i);
+                   
+        my $indexFoundLicence;
+        if ($object->{'ID'} && $seenProducts->{lc($object->{'ID'})}) {
+            my $seenKey = $seenProducts->{lc($object->{'ID'})};
+            if ($seenKey->{'/ProductCode'}) {
+                my $ProductCodeUuid =  lc( $seenKey->{'/ProductCode'} =~ /([-\w]+)/ && $1 );
+                # Check Office registry is true
+                if ($seenProducts->{$ProductCodeUuid} && $seenProducts->{$ProductCodeUuid}->{'/DigitalProductID'}) {
+                    # Find Office registry licence
+                    my $passLicence = 0;
+                    foreach my $seenLicence (@{$licences}) {
+                        if($seenLicence->{PRODUCTID} eq $seenProducts->{$ProductCodeUuid}->{'/ProductID'}) {
+                            $passLicence = 1 if $seenLicence->{KEY} =~ m/$partialProductKey$/;
+                            # $passLicence = 0, so the registry calculation of the key is wrong, WMI information are more revelant, so we delete the wrong Licence Office Registry
+                            ($indexFoundLicence) = grep { @{$licences}[$_] eq $seenLicence } 0..$#$licences if !$passLicence;
+                            last;
+                        }
+                    }
+                    # $passLicence = 1, Registry Licence is good, it supply more information, so we skip WMI Office Licence
+                    next if $passLicence;
+                }
+                # Update FULLNAME if seen ProductName in registry
+                $license->{FULLNAME} = encodeFromRegistry($seenKey->{'/ProductName'})
```

Comment:
Here, I know you just move an existing code, but do you think the encodeFromREgistry() call is necessary. I think it still has been called the first time at L.162-163. Can you fix it and validate it in your case ?